jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/HttpOutput.java

```diff
@@ -1096,6 +1149,7 @@ else if (crlf != null && crlf.hasRemaining())
      * @param content The whole content to send
      * @throws IOException if the send fails
      */
+    @Deprecated(forRemoval = true, since = "12.1.0")
```

Comment:
Must add also the `@deprecated` javadoc tag.
Why are they deprecated?

Reply:
But application were able to cast to `HttpOutput` and use these methods, for example if they had a `ByteBuffer` and did not want to use the `OutputStream` APIs.

Do we offer an alternative? If so, should be documented in the javadocs.

Reply:
> Also I doubt any applications were down casting like this.

Our own FastFileServer demo/embedded does that.
And it has been copied to other projects for years now.

Some examples of applications down casting like that ...

* https://github.com/hamann/pedestal/blob/ecae8023662b79b95e7b4df6df9de7c130ae0aff/jetty/src/io/pedestal/http/jetty/container.clj#L19-L31
* https://github.com/apache/ignite-extensions/blob/a51fb6d0e9c1129bcf190990687134bea1efb90a/modules/mesos-ext/src/main/java/org/apache/ignite/mesos/resource/ResourceHandler.java#L125-L127
* https://github.com/haoxianrui/spring-framework/blob/e5163351c47feb69483e79fa782eec3e4d8613e8/spring-web/src/main/java/org/springframework/http/server/reactive/JettyHttpHandlerAdapter.java#L128-L131
* https://github.com/cognitedata/bazel-maven-proxy/blob/4360308592b4a4dd4165482281ecbfd08a257636/server/src/main/java/com/salesforce/bazel/maven/proxy/server/MavenProxyServlet.java#L375-L392

Reply:
Hmmmm then we'd better add tests for them.

Reply:
@joakime I've updated HttpOutputTest for EE10 and EE11